Lib/asyncio/proactor_events.py

```diff
@@ -244,7 +271,15 @@ def _eof_received(self):
         if not keep_open:
             self.close()
 
-    def _data_received(self, data, length):
+    def _buffer_updated(self, length):
+        try:
+            self._protocol.buffer_updated(length)
+        except BaseException as exc:
+            self._fatal_error(exc,
+                              'Fatal error: protocol.buffer_updated() '
+                              'call failed.')
+
+    def _buffer_updated(self, length):
```

Comment:
bug, this function is defined twice now

Reply:
Well it worked but just defined twice,
Fixed.